sdk/core/Azure.Core/tests/PageableHelpersTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests
+{
+    public class PageableHelpersTests
+    {
+        [Test]
+        public void PageableHelpers_CreateAsyncPageableFromDelegates()
+        {
+            var cts = new CancellationTokenSource();
+            var enumerable = CreateAsyncPageableFromDelegates(cts).WithCancellation(cts.Token);
+
+            Assert.CatchAsync<OperationCanceledException>(async () =>
+            {
+                await foreach (var i in enumerable) { }
+            });
+        }
+
+        [Test]
+        public void PageableHelpers_CreateAsyncPageableFromEnumerable()
```

Comment:
In this test, `AsyncPageable` is created from `IAsyncEnumerable` which passes `CancellationToken` into enumerator.